Parser/Token.cs

```diff
@@ -38,9 +39,32 @@ public enum E {
    // Utility function used to echo an error to the console
    public void PrintError () {
       if (Kind != ERROR) throw new Exception ("PrintError called on a non-error token");
-      Console.ForegroundColor = ConsoleColor.Yellow;
-      Console.WriteLine ($"At line {Line}, column {Column} of {Source.FileName}: {Text}");
-      Console.ResetColor ();
+      string hdr = $"File: {Source.FileName}";
+      OutputEncoding = Encoding.Unicode;
+      Out (hdr);
+      Out (new string ('\u2501', hdr.Length));
+      int ln = 0, start = 4, err = Column - 1 + start;
+      for (int i = 3; i > 0; i--) {
+         if ((ln = Line - i) < 0) continue;
+         Out (Source.Lines[ln], ln + 1);
+      }
+      ForegroundColor = ConsoleColor.Yellow;
+      Out (new string (' ', err) + '^');
+      int mid = Text.Length / 2 - (Text.Length % 2 == 0 ? 1 : 0);
+      int left = err - mid;
+      CursorLeft = left > start ? left : start;

```

Comment:
Instead of setting CursorLeft, just pad your output with the required number of spaces. Then, it will render correctly even when we are redirecting the output to a file.